### PR TITLE
Add `docker-` prefix to runc and containerd binaries

### PR DESCRIPTION
**Pull request by tiborvass**
Signed-off-by: Tibor Vass tibor@docker.com

### CODE REVIEW

**Comment by icecrime** [comment]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -253,19 +253,23 @@ RUN set -x \
...
+  	&& git clone git://github.com/opencontainers/runc.git "$GOPATH/src/github.com/opencontainers/runc" \
 	&& cd "$GOPATH/src/github.com/opencontainers/runc" \
 	&& git checkout -q "$RUNC_COMMIT" \
-	&& make BUILDTAGS="seccomp apparmor selinux" && make install
+	&& make static BUILDTAGS="seccomp apparmor selinux" \
+	&& cp runc /usr/local/bin/docker-runc
 
 # Install containerd
-ENV CONTAINERD_COMMIT 7146b01a3d7aaa146414cdfb0a6c96cfba5d9091
+ENV CONTAINERD_COMMIT 142e22a4dce86f3b8ce068a0b043489d21976bb8
```
Comment (mlaventure) [inline Dockerfile]: It shouldn't matter, but can this be set into `hack/vendor.sh` too?

**Comment by tonistiigi** [comment]
LGTM

**Comment by mlaventure** [comment]
Looks like hub reverted their update, the `DockerSuite.TestPushToCentralRegistryUnauthorized` can be ignored.

Regarding `docker_cli_run_unix_test.go:852: DockerSuite.TestRunSeccompDefaultProfile`, the node went out of space. Since it passed on the other tests, I'd vote to ignore it too ;-)

**Comment by mlaventure** [comment]
Tested locally, SGTM.

**Comment by runcom** [comment]
I don't understand this change, I don't usually use `make` from the Makefile to compile and install docker+deps. With this even if I manually compile containerd and runc I have to `mv` the binaries to docker-*. It's a bad experience.

**Comment by tianon** [comment]
I agree with @runcom and @mlaventure -- if we're shipping these as "Docker" artifacts, they should be vendored and built from the Docker source.  What's the expected workflow ATM for packagers to handle this bit?  They'll already be packaging containerd, so should they be patching the source to remove this expected `docker-` prefix?  Should their packages supply some kind of symlink with `docker-` in the name to satisfy Docker's expectation?

As a separate but related note, why _are_ we shipping these as Docker artifacts instead of relying on the fact that they are separate projects with separate release cycles?  Is this specific commit going to be "the commit" of each of those projects which is intended to work properly with Docker 1.11.0, and any mismatch may cause unexpected breakage?

**Comment by jfrazelle** [comment]
I agree with @tianon and @runcom I think from a packager standpoint its really hard to change the binary for a runc dependency package to be named docker-runc, or for them to patch docker which really we shouldnt want at all

**Pull request merged**